### ISSUE TITLE
[6.1.1] reset() changes the form to valid even if it is not (possible bug)

### ISSUE THREAD

**Issue by Mathius17** [erikras/redux-form#2067]
I have a simple form with 2 inputs and 2 buttons, one to `submit` the form, and the other one to `reset` the form. Both of this inputs are `invalid` if they are empty.

Steps for reproduction:
1. `submit` the form (get 2 errors)
2. Press the `reset button` and both errors disappear
3. `submit` the form again, it suddenly passes validation (even if it doesn't, inputs are empty) and proceeds to handle the submit!

As suggested in other issues, the problem may be solved if `UPDATE_SYNC_ERRORS ` is executed after `reset` and `destroy`.

Possibly related to #1776, #1902, #1385, #1865

**Comment by JakeLin1024** [comment]
Hi, @Mathius17 
What kind of form validation are you using? Syn, asyn or submit. Do you mind sharing your code?

**Comment by JakeLin1024** [comment]
btw, I dont have this issue using submit validation in v6.1.1

**Comment by Mathius17** [comment]
Hi, JakeLin1024

I'm not sure if I'm using sync or submit validation, but here's the code:
```javascript
import React from 'react';
import { Field, reduxForm } from 'redux-form';

const validate = values => {
  const errors = {};

  if (!values.email) {
    errors.email = "Ingresa un email";
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Ingresa un email válido'
  }

  if (!values.password) {
    errors.password = "Ingresa la contraseña";
  }

  return errors;
};

class Login extends React.Component {

  handleSubmit = (values) => {
    console.log('submitted!');
    // this.props.actions.login(values.email, values.password);
  }

  renderField = ({ input, label, type, meta: { touched, error } }) => (
    <fieldset className={`input-group ${touched && error ? 'has-error' : ''}`}>
      <label>{label}</label>
      <input {...input} type={type} placeholder={label} />
      {touched && error && <div className="error-block">{error}</div>}
    </fieldset>
  );

  render = () => (
    <div className="container login">
      <form className="row" onSubmit={this.props.handleSubmit(this.handleSubmit)}>
        <div className="column column-40">
          <Field name="email" component={this.renderField} type="email" label="Email"/>
          <Field name="password" component={this.renderField} type="password" label="Contraseña"/>
          <button type="submit" className="button-primary">Login</button>
          <button type="button" className="button-primary" onClick={() => this.props.reset()}>
            Reset
          </button>
        </div>
      </form>
    </div>
  );
}

export default reduxForm({
  form: 'login',
  validate
})(Login);
```

I'll try to post a gif to show the problem.

**Comment by Mathius17** [comment]
Here's the redux inspector log

<img width="290" alt="screenshot 2016-11-03 07 28 17" src="https://cloud.githubusercontent.com/assets/6528566/19965912/44aff798-a197-11e6-898f-60178642bda6.png">

**Comment by piotrkl** [comment]
Is there any progress on this issue?

**Comment by kmanishprdxn** [comment]
Is there any update on this. **I'm using redux-form@6.1.1 and facing same problem.**

When I click on submit button handleSubmit validates all the fields and give corresponding error message but if I reset the form first and then click on submit button, handleSubmit skip the validation and call provided callback/submit function, which is incorrect.

**Comment by kmanishprdxn** [comment]
@Mathius17 @piotrkl any updates or any solution if you have.

**Comment by stevenmusumeche** [comment]
I'm having the same issue.

**Comment by kmanishprdxn** [comment]
Hi All,

This issue has been fixed with latest release. refer #2244, #2246

I'm using **v6.3.2** and now onSubmit validation is working as intended even after resting the form.

**Comment by gustavohenke** [comment]
Hi @Mathius17, can you please update Redux Form and check whether @kmanishprdxn is correct?

**Comment by Mathius17** [comment]
@gustavohenke sorry guys, I'm on vacation and I won't be near my computer till next Thursday. But if my memory is correct, I recently updated the redux-form dependency and I wasn't having this problem.

**Comment by Mathius17** [comment]
This was fixed some releases ago (6.3+), if it continues for you feel free to keep discussing it
